lib/internal/crypto/scrypt.js

```diff
@@ -107,8 +109,13 @@ function check(password, salt, keylen, options) {
       p = options.parallelization;
     }
     if (options.maxmem !== undefined) {
-      validateUint32(options.maxmem, 'maxmem');
```

Comment:
I think you could replace this with `validateInteger()`. That checks `typeof` and `Number.isSafeInteger()`. You'd only need an additional `maxmem < 0` check I think (although some of the other validators support min and max values, and that functionality could be added).

Reply:
I considered that, but `validateInteger(2 ** 53)` would claim that the value is not "an integer", despite `Number.isInteger(2 ** 53) === true`. Personally, I think the current behavior is better, which identifies `2 ** 53` as an out-of-range integer (instead of "not an integer"). Also, this preserves the error messages. What do you think?

Reply:
> What do you think?

I think we should use `validateInteger()` for consistency, code reuse, and code coverage. You make valid points though. I update `validateInteger()` in https://github.com/nodejs/node/pull/28810. I think `validateInteger(maxmem, 'maxmem', 0)` would accomplish the same thing as this code block now.